src/Shared/Sdk/Providers/ServiceClients/ClientFactory.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using ACMTTU.NoteSharing.Shared.DataContracts;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Newtonsoft.Json;
+
+namespace ACMTTU.NoteSharing.Shared.SDK.Clients
+{
+    public class ClientFactory : IClientFactory
+    {
+        // Name of the Kubernetes Service associated with the Secrets Service
+        private const string secretServiceName = "secretsservice";
+
+        // Name of the Kubernetes namespace the Secrets Service is running on
+        private const string secretServiceNamespace = "secrets";
+
+        private readonly IHttpClientFactory factory;
+
+        public ClientFactory(IHttpClientFactory factory)
+        {
+            this.factory = factory;
+        }
+
+        /// <summary>
+        /// Gets the Database Client for that namespace
+        /// </summary>
+        /// <returns></returns>
+        public CosmosClient GetDatabaseClient(string connectionString)
+        {
+            return new CosmosClient(connectionString);
+        }
+
+        /// <summary>
+        /// Gets the Storage Client for that namespace
+        /// </summary>
+        /// <returns></returns>
+        public CloudBlobClient GetStorageClient(string connectionString)
+        {
+            // Get the storage account
+            CloudStorageAccount storageAccount;
+            if (!CloudStorageAccount.TryParse(connectionString, out storageAccount))
+            {
+                throw new Exception($"Could not parse connection string: {connectionString}");
+            }
+
+            // Create the client for the storage account
+            return storageAccount.CreateCloudBlobClient();
+        }
+
+        /// <summary>
+        /// Calls the Secrets Service to grab the specified connection strings
+        /// </summary>
+        /// <param name="option">Your choice of whether you want the Database or the Storage Client</param>
+        /// <returns></returns>
+        public async Task<string> GetConnectionStringForClient(ClientOptions option)
+        {
+            string connectionString;
+
+            HttpClient client = this.factory.CreateClient();
```

Comment:
Without a using statement the HttpClient returned by this factory will never close unless explicitly closed. The factory will return a new instance each time creating a memory leak.

Reply:
I am away of the IDispoable interface and that we do use using...

I actually did some research on this and there are a few things I found

https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/

TL;DR look at the bottom of the article

Reply:
https://stackoverflow.com/questions/15705092/do-httpclient-and-httpclienthandler-have-to-be-disposed

Another one.

Reply:
I see... How strange. In that case we need to ensure we use a singleton service, not a transient one, when using DI.